File: cases/taint/php/element-object-negative/element_object_negative.php
<?php
class Item
{
    public string $value = "clean";
}

function dfb_source(): string { // DFB-SOURCE: element-object-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: element-object-sink

function run(): void {
    $items = [new Item(), new Item()];
    $items[0]->value = dfb_source(); // DFB-WITNESS: element-object-store
    $items[1]->value = "clean";
    dfb_sink($items[1]->value);
}

File: cases/taint/csharp/element-object-negative/ElementObjectNegative.cs
namespace DataFlowBench;

static class ElementObjectNegative
{
    sealed class Item
    {
        public string Value = "clean";
    }

    static string dfb_source() // DFB-SOURCE: element-object-input
    {
        return "tainted";
    }

    static void dfb_sink(string value) { } // DFB-SINK: element-object-sink

    static void Run()
    {
        Item[] items = new Item[] { new Item(), new Item() };
        items[0].Value = dfb_source(); // DFB-WITNESS: element-object-store
        dfb_sink(items[1].Value);
    }
}
